### PR TITLE
PWA 12.5 docs

### PR DESCRIPTION
**Pull request by dobooth**
## Description

This PR contains content for the PWA 12.5 release, focusing on the new eventing framework.

## Affected Pages

https://developer.adobe.com/commerce/pwa-studio/integrations/analytics/
https://developer.adobe.com/commerce/pwa-studio/integrations/analytics/custom-events
https://developer.adobe.com/commerce/pwa-studio/integrations/analytics/event-reference
https://developer.adobe.com/commerce/pwa-studio/integrations/analytics/installation-and-configuration

## Issue/Ticket Number

[PWA](https://jira.corp.magento.com/browse/PWA-2365)

### CODE REVIEW

File: src/pages/integrations/analytics/custom-events/index.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+---
+
+# Creating custom events
+
+PWA Studio 12.5 contains an eventing framework with a series of default events that cover common scenarios.
+By creating and importing a custom extension, it is possible to add new events so that you may track specific user interactions beyond what is included in the default package.
+
+PWA Studio provides a [sample extension](https://github.com/magento/pwa-studio/tree/develop/packages/extensions/experience-platform-connector/) that shows how to add a custom event.
+
+In the sample extension, the `SEARCH_REPSONSE` event is being tracked. This event is fired when a search result is returned from the Adobe Experience Manager (AEM), with either results or a `EMPTY_RESULT` response.
```
Comment (jcalcaben) [inline src/pages/integrations/analytics/custom-events/index.md:12]: * I hope it's not spelled `SEARCH_REPSONSE` in the codebase 😬 
* the event is fired when Magento returns search results data, not AEM

File: src/pages/integrations/analytics/custom-events/index.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+These files can be used as a starting point for a custom event extension.
+
+## searchResponseReceived.js
+
+This file sets the payload and the search results context. The custom event name is also defined in this file.
```
Comment (jcalcaben) [inline src/pages/integrations/analytics/custom-events/index.md:18]: Seems to be a lot of relevant code missing from this snippet.

Also, I would describe this file as an event handler file that returns two functions:

1. `canHandle()` - a function that receives an event from PWA Studio's eventing framework and returns true if the `handle()` function in this file should handle the incoming event
2. `handle()` - function that handles the event

File: src/pages/integrations/analytics/custom-events/index.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
+---
+title: Custom events for Beacon analytics
```
Comment (jcalcaben) [inline src/pages/integrations/analytics/custom-events/index.md:2]: the official name is Experience Platform(I think), not Beacon

File: src/pages/integrations/analytics/custom-events/index.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    };
+
+    sdk.context.setSearchResults(searchResultsContext);
+    sdk.publish.searchResponseReceived(searchUnitId, searchResultsContext);
+};
+```
+
+## wrapUseAutocomplete.js
+
+In `wrapUseAutocomplete.js` the search result is received and the payload defined and dispatched. This take place in the `useEffect` function:
```
Comment (jcalcaben) [inline src/pages/integrations/analytics/custom-events/index.md:60]: Would describe this file as a talon wrapper that wraps around the `useAutocomplete()` talon to add event dispatching logic.

Would link to doc page about [talons](https://developer.adobe.com/commerce/pwa-studio/guides/general-concepts/talons/), tutorial on [how to wrap a talon](https://developer.adobe.com/commerce/pwa-studio/tutorials/targets/modify-talon-results/), and whatever docs we have about the eventing framework.

File: src/pages/integrations/analytics/custom-events/index.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+```javascript
```
Comment (jcalcaben) [inline src/pages/integrations/analytics/custom-events/index.md:62]: Missing a lot of relevant code

File: src/pages/integrations/analytics/custom-events/index.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                        searchUnitId: 'search-bar',
+                        suggestions: displayResult || []
+                    }
+                });
+            }
+```
+
+## intercept.js
+
+Edit copy of `intercept.js` to pull in the custom extension by name. In this case, the `SearchBar` wrapper:
```
Comment (jcalcaben) [inline src/pages/integrations/analytics/custom-events/index.md:86]: This is an intercept file that interacts with PWA Studio's extensibility framework to modify code during build time.
In this particular code, the `useApp()`, `useAccountMenu()`, and `useAutocomplete()` talons are wrapped with modules that provide additional logic.

Would link to:
* [docs on extensibility](https://developer.adobe.com/commerce/pwa-studio/guides/general-concepts/extensibility/)
* [modify talons tutorial](https://developer.adobe.com/commerce/pwa-studio/tutorials/targets/modify-talon-results/)

File: src/pages/integrations/analytics/custom-events/index.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        App.useApp.wrapWith('@magento/experience-platform-connector');
+        Header.useAccountMenu.wrapWith(
+            '@magento/experience-platform-connector/src/wrappers/wrapUseAccountMenu'
+        );
+        SearchBar.useAutocomplete.wrapWith(
+            '@magento/experience-platform-connector/src/wrappers/wrapUseAutocomplete'
+        );
+```
+
+## Import a custom extension
```
Comment (jcalcaben) [inline src/pages/integrations/analytics/custom-events/index.md:99]: Not sure this is needed. Most developers should already know how to set up a new package but if you really want to start from scratch, I would refer to the section in this tutorial on [how to initialize a project](https://developer.adobe.com/commerce/pwa-studio/tutorials/targets/modify-talon-results/#initialize-the-project).

I would replace this particular section with a testing locally section that looks like [this](https://developer.adobe.com/commerce/pwa-studio/tutorials/targets/modify-talon-results/#test-on-a-local-instance)

**Review by jcalcaben — CHANGES_REQUESTED** [review]
Is there a topic that describes the core eventing framework that the analytics extension taps into?

File: src/pages/integrations/analytics/custom-events/index.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+# Creating custom events
+
+PWA Studio 12.5 contains an eventing framework with a series of default events that cover common scenarios.
+By creating and importing a custom extension, it is possible to add new events so that you may track specific user interactions beyond what is included in the default package.
+
+PWA Studio provides a [sample extension](https://github.com/magento/pwa-studio/tree/develop/packages/extensions/experience-platform-connector/) that shows how to add a custom event.
+
+In the sample extension, the `SEARCH_RESPONSE` event is being tracked. This event is fired when a search result is returned from Adobe Commerce, with either results or a `EMPTY_RESULT` response.
+
+These files can be used as a starting point for a custom event extension.
```
Comment (bdenham) [inline src/pages/integrations/analytics/custom-events/index.md:14]:
```suggestion
These files can be used as a starting point to create your own custom events.
```

File: src/pages/integrations/analytics/custom-events/index.md (hunk 5 above)
Comment (bdenham) [reply src/pages/integrations/analytics/custom-events/index.md:70]: Deferring to James.

File: src/pages/integrations/analytics/index.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
+---
+title: Experience Platform analytics in PWA Studio
+---
+
+PWA Studio supports analytics through a built-in framework that sends events to your Adobe Commerce backend.
+The following topics describe how to install, configure and customize analytics to your specific needs.
```
Comment (bdenham) [inline src/pages/integrations/analytics/index.md:6]:
```suggestion
The following topics describe how to install, configure, and customize analytics to your specific needs.
```

**Review by bdenham — CHANGES_REQUESTED** [review]
See changes.

File: src/pages/integrations/analytics/index.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
+---
+title: Experience Platform analytics in PWA Studio
+---
+
```
Comment (bdenham) [inline src/pages/integrations/analytics/index.md:4]:
```suggestion

# Experience Platform analytics in PWA Studio

```

File: src/pages/guides/general-concepts/analytics/index.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
+---
+title: Analytics Events
+---
+
+## PWA analytics
```
Comment (bdenham) [inline src/pages/guides/general-concepts/analytics/index.md:5]:
```suggestion
# Analytics events
```

**Review by bdenham — APPROVED** [review]
Couple more changes. Approved on edit.

File: src/pages/tutorials/events/custom-events/index.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
+---
+title: Custom events for Experience Platform analytics
+---
+
+# Creating custom events
+
+PWA Studio 12.5 adds an eventing framework to capture a set of default, everyday shopper interactions on your storefronts.
```
Comment (jcalcaben) [inline src/pages/tutorials/events/custom-events/index.md:7]:
```suggestion
The eventing framework introduced in the PWA Studio 12.5 release comes with a core set of storefront events related to various interactions with the application.
```

File: src/pages/tutorials/events/custom-events/index.md [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+You can also extend the framework by adding additional events to track user interactions beyond those in the default package.
```
Comment (jcalcaben) [inline src/pages/tutorials/events/custom-events/index.md:8]:
```suggestion
You can use PWA Studio's extensibility framework to modify your storefront and have it publish events not covered by the default set.
```

File: src/pages/tutorials/events/custom-events/index.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+
+PWA Studio provides a [sample extension](https://github.com/magento/pwa-studio/tree/develop/packages/extensions/experience-platform-connector/) that shows how to add a custom event.
```
Comment (jcalcaben) [inline src/pages/tutorials/events/custom-events/index.md:10]:
```suggestion
PWA Studio provides the source for the [Experience Platform Connector extension](https://github.com/magento/pwa-studio/tree/develop/packages/extensions/experience-platform-connector/) to show how you can add a custom event.
```

File: src/pages/tutorials/events/custom-events/index.md [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+
+In this sample extension, we added a `SEARCH_RESPONSE` event. This event fires when a search result or an `EMPTY_RESULT` response is returned from Adobe Commerce.
```
Comment (jcalcaben) [inline src/pages/tutorials/events/custom-events/index.md:12]:
```suggestion
This extension dispatches a `SEARCH_RESPONSE` event, which is not part of the default set. This event fires when a search result is returned in the application's search bar.
```

**Review by jcalcaben — CHANGES_REQUESTED** [review]

**Review by jcalcaben — APPROVED** [review]

**Pull request merged**